var/spack/repos/builtin/packages/dpcpp/package.py

```diff
@@ -16,80 +17,122 @@ class Dpcpp(CMakePackage):
     git = "https://github.com/intel/llvm.git"
 
     version("develop", branch="sycl")
-    version("2021.09", commit="bd68232bb96386bf7649345c0557ba520e73c02d")
-    version("2021.12", commit="27f59d8906fcc8aece7ff6aa570ccdee52168c2d")
+    version(
+        "2023-03",
+        sha256="ca85303d712c58316a91a7c97f7c78fa563a29f1669d8b2368d0c8bd92a63068",
+        url="https://github.com/intel/llvm/tarball/cb91c232",
+    )
+    version("2021.09", commit="bd68232bb96386bf7649345c0557ba520e73c02d", deprecated=True)
+    version("2021.12", commit="27f59d8906fcc8aece7ff6aa570ccdee52168c2d", deprecated=True)
 
     maintainers("ravil-mobile")
+
     variant("cuda", default=False, description="switch from OpenCL to CUDA")
-    variant("rocm", default=False, description="switch from OpenCL to ROCm")
+    variant("hip", default=False, description="switch from OpenCL to HIP")
```

Comment:
Why changing name to a variant that is widely used throughout Spack?

Reply:
I changed the name to HIP since DPC++ supports both HIP AMD and HIP NVIDIA backends. Also the [buildbot](https://github.com/intel/llvm/blob/sycl/buildbot/configure.py) script I used as a reference renamed it from rocm to hip. 

Although, DPC++ recommends not to build it with HIP NVIDIA and just prefer CUDA. So we could change it back to rocm if its preferred.

Reply:
HIP nvidia is unfortunately a joke. It was generated by AMD to try to make their compiler seem cross platform, but in reality every scientific app has to write another compatibility layer of their own to work with HIP and CUDA. Not even Cmake supports using the nvidia backend for hip. It's best to look at HIP as a cuda like layer that AMD added to rocm. Please don't change the variant name.

Reply:
https://docs.olcf.ornl.gov/systems/summit_user_guide.html#hip-code-cannot-be-built-using-cmake-using-hip-host-device-or-hip-language-support

Reply:
Yeah from looking at other scientific applications it seems like a common thing to just avoid HIP NVIDIA. I'll change back to the rocm variant.

Reply:
If I change the variant back to `rocm`, what should I do with the `hip-platform` variant? Should I assume that we always we want to build the HIP AMD backend and ignore the HIP NVIDIA backend?